buildSrc/src/main/java/org/librarysimplified/gradle/RequiredAssetsTask.kt

```diff
@@ -0,0 +1,94 @@
+package org.librarysimplified.gradle
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.security.MessageDigest
+
+/**
+ * Returns the SHA-256 message digest of the given file.
+ */
+fun sha256(file: File): String {
+  return MessageDigest.getInstance("SHA-256")
+    .digest(file.readBytes())
+    .fold("", { str, it -> str + "%02x".format(it) })
+}
+
+/**
+ * Ensure the required files are present in the assets/ directory of
+ * the assembled APK.
+ */
+open class RequiredAssetsTask : DefaultTask() {
+
+  /** A map of filenames to a sha256 hash. */
+  @get:Input
+  var required: Map<String, String> = emptyMap()
+
+  /** A list of input APKs to check for the required files. */
+  @get:InputFiles
+  var apkFiles: List<File> = emptyList()
+
+  companion object {
+    const val STRICT_MODE = "org.librarysimplified.strictmode"
+  }
+
+  private val strictMode by lazy {
+    project.findProperty(STRICT_MODE)?.toString()?.toBoolean() ?: false
```

Comment:
I'm not sure I like this being off by default, and this was one of the major reasons that SimplyE was in a separate repos: Someone building SimplyE in that other repos is _required_ to have set everything up properly and the build will fail early if they haven't. This new setup would seem to allow them to build a broken application with missing files and then find out about it at runtime.

Reply:
Followed up on this below!